gtest/src/program.rs

```diff
@@ -563,9 +566,11 @@ impl<'a> Program<'a> {
         fn_name: &str,
         wasm: Vec<u8>,
         argument: Option<E>,
+        payload: Vec<u8>,
```

Comment:
Better to make `payload` the first argument. Also, one should consider to use `P: Encode` for `payload`.